File: code/smv-debugger/src/main/java/org/fbme/smvDebugger/integration/ServicePathProvider.java
package org.fbme.smvDebugger.integration;

import jetbrains.mps.project.MPSProject;
import org.jetbrains.annotations.NotNull;

import java.nio.file.Path;

public interface ServicePathProvider {

    Path getFb2SmvServicePath();

    Path getNuSmvServicePath();

    Path getNutracServicePath();

    abstract class Factory {
        protected static ServicePathProvider.Factory INSTANCE = null;

        protected abstract ServicePathProvider create(@NotNull MPSProject project);
    }

    static ServicePathProvider create(@NotNull MPSProject project) {
        return Factory.INSTANCE.create(project);
    }
}

File: code/smv-debugger/src/main/kotlin/org/fbme/smvDebugger/integration/ServicePathProvider.kt
package org.fbme.smvDebugger.integration

import jetbrains.mps.project.MPSProject
import java.nio.file.Path

interface ServicePathProvider {
    val fb2SmvServicePath: Path?
    val nuSmvServicePath: Path?
    val nutracServicePath: Path?

    abstract class Factory {
        abstract fun create(project: MPSProject): ServicePathProvider

        companion object {
            var INSTANCE: Factory? = null
        }
    }

    companion object {
        fun create(project: MPSProject): ServicePathProvider {
            return Factory.INSTANCE!!.create(project)
        }
    }
}
